views/dapp/v2/components/layout/header/menu/settings/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { Button, Theme, useTheme } from '@interest-protocol/ui-kit';
+import { useRouter } from 'next/router';
+import { FC, useState } from 'react';
+
+import { CogsSVG } from '@/components/svg/v2';
+import { RefBox } from '@/elements';
+import { useLocale } from '@/hooks';
+import useClickOutsideListenerRef from '@/hooks/use-click-outside-listener-ref';
+
+import MenuLanguage from './menu-language';
+import MenuSettings from './menu-settings';
+
+const BOX_ID = 'lang-switch-box-id-123';
```

Comment:
Choose mnemonic IDs related to what you're doing with the ref.

Eg: BOX_ID = 'settings-dropdown'